week3/createContentTrainingData.py

```diff
@@ -3,10 +3,16 @@
 import random
 import xml.etree.ElementTree as ET
 from pathlib import Path
+from nltk import SnowballStemmer
+import re
 
+stemmer = SnowballStemmer("english")
 def transform_name(product_name):
-    # IMPLEMENT
-    return product_name
+    # IMPLEMENTED by Davide Del Vecchio
+    #trasformed using the snowball stemmer and substituting the 
+    product_name_clean=re.sub(r'''\W+\s*''', ' ', product_name)
+    product_name_stemmed = stemmer.stem(product_name_clean)
```

Comment:
I like the use of the snowball stemmer.  I think the `stem()` method requires the input to be split into words to be most effective.